docs/troubleshooting.md

```diff
@@ -0,0 +1,83 @@
+| shuffle server storage audit log | rss.server.storage.audit.log.enabled  | false   | The server will log audit records for every disk write and delete operation |
+
+Based on the above audit logs, you can check the operation details and the operation time cost.
+
+## Uniffle remote debug
+
+### Debugging Uniffle processes
+
+Java remote debugging makes it easier to debug Uniffle at the source level without modifying any code. You will need to set the JVM remote debugging parameters before starting the process. There are several ways to add the remote debugging parameters; you can export the following configuration properties in shell or conf/rss-env.sh:
+
+```shell
+# Java 8
+export DASHBOARD_JAVA_OPTS="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5004"
+export COORDINATOR_JAVA_OPTS="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5006"
+export SHUFFLE_SERVER_JAVA_OPTS="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5005"
+
+# Java 11
+export DASHBOARD_JAVA_OPTS="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=*:5004"
+export COORDINATOR_JAVA_OPTS="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=*:5006"
+export SHUFFLE_SERVER_JAVA_OPTS="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=*:5005"
+```
+
+In general, you can use `<PROCESS>_JAVA_OPTS` to specify how an Uniffle process should be attached to.
+
+suspend={y | n} will decide whether the JVM process waits until the debugger connects or not.
+
+address determines which port the Uniffle process will use to be attached to by a debugger. If left blank, it will choose an open port by itself.
+
+After completing this setup, learn how [To attach](#to-attach).
+
+### To attach
+
+There exists a [comprehensive tutorial on how to attach to and debug a Java process in IntelliJ](https://www.jetbrains.com/help/idea/attaching-to-local-process.html).
+
+Start the process or a shell command of interest, then create a new java remote configuration,
+set the debug server's host and port, and start the debug session. If you set a breakpoint which can be reached, the IDE
+will enter debug mode. You can inspect the current context's variables, call stack, thread list, and expression
+evaluation.
+
+## Resource Leak Detection
+
+If you are operating your Uniffle cluster it is possible you may notice a message in the logs like:
+
+```
+[ERROR] ResourceLeakDetector - LEAK: ByteBuf.release() was not called before it's garbage-collected. See https://netty.io/wiki/reference-counted-objects.html for more information.
+```
+
+Uniffle has a built-in detection mechanism to help identify potential resource leaks. This message indicates there is a bug in the Uniffle code which is causing a resource leak. If this message appears during cluster operation, please open a GitHub Issue as a bug report and share your log message and any relevant stack traces that are shared with it.
```

Comment:
Uniffle uses Netty's built-in memory leak detection mechanism to help identify potential resource leaks. This message implies that there might be a bug in the Uniffle code, causing a resource leak. If this message appears while the cluster is running, please open a GitHub Issue as a bug report and share your log message, along with any relevant stack traces associated with it.